sdk/core/core-http/CHANGELOG.md

```diff
@@ -2,10 +2,12 @@
 
 ## 1.1.7 (Unreleased)
 
-
 ## 1.1.6 (2020-08-04)
 
 - Update dependencies `@azure/core-tracing` to version 1.0.0-preview.9 and `@opentelemetry/api` to version 0.10.2 [PR 10393](https://github.com/Azure/azure-sdk-for-js/pull/10393)
+- `BearerTokenAuthenticationPolicy` constructor signature has been updated to accommodate the token refresher appropriately. The signature for `bearerTokenAuthenticationPolicy` method doesn't change.
+  [PR 10728](https://github.com/Azure/azure-sdk-for-js/pull/10728)
+  Note: It is expected that the users depending on `BearerTokenAuthenticationPolicy` do so indirectly through `bearerTokenAuthenticationPolicy` method that is exported.
```

Comment:
This feels like a lot of context/details for a change to what is effectively an internal type.

```suggestion
- Fixed a bug in `BearerTokenAuthenticationPolicy` where a helper class was being instantiated on each request. [PR 10728](https://github.com/Azure/azure-sdk-for-js/pull/10728)
```

I think this is enough honestly. Nobody should have a reason to grab the per-instance class and mess with it. Alternatively we could call this out as 'BREAKING' but honestly `BearerTokenAuthenticationPolicy` should be private. 

I don't really understand why we exported that type, since it's not used as the return type of `bearerTokenAuthenticationPolicy` -- other policy classes aren't exported e.g.

```ts
export { LogPolicyOptions, logPolicy } from "./policies/logPolicy";
```

But some recent ones like `DisableResponseDecompressionPolicy` are, which just feels like a bug. Maybe this was done to make tests easier to write?

Reply:
> I don't really understand why we exported that type, since it's not used as the return type of bearerTokenAuthenticationPolicy -- other policy classes aren't exported

Hmmm, they shouldn't have been exported.

> But some recent ones like DisableResponseDecompressionPolicy are, which just feels like a bug. 

Looks like KeepAlivePolicy is being exported too.
`export { keepAlivePolicy, KeepAlivePolicy, KeepAliveOptions } from "./policies/keepAlivePolicy";`


> Maybe this was done to make tests easier to write?

We import the src code directly inside the tests, no need to export it through `index.ts`(or `coreHttp.ts`) just for the sake of tests.

Reply:
Opened a draft PR https://github.com/Azure/azure-sdk-for-js/pull/10738 to unexport these classes.
I'll wait for the build to succeed before moving it out of the draft!